lib/OpusVL/FB11/Auth/Brain.pm

```diff
@@ -22,18 +55,127 @@ has short_name => (
     default => 'fb11-auth',
 );
 
+=head1 HATS AND SERVICES
+
+=head2 auth
+
+This Brain wears an auth hat and provides the auth service. See
+L<OpusVL::FB11::Auth::Brain::Hat::auth>.
+
+This is essentially a stand-in for a future auth service that has not been
+designed yet.
+
+=head2 fb11authdb
+
+This is an alias into the auth hat and service.
+
+=head2 dbicdh::manager
+
+The best way to ensure your schemata get deployed is to have this brain in your
+hive, since it provides the C<dbicdh::manager> service.
+
+Remember that nothing actually accesses this service by default: this is
+because, in principle, FB11 should not require a database to run.
+
+=head2 dbicdh::consumer
+
+The FB11AuthDB is deployed and upgraded as the zeroth priority schema, and thus
+is both the manager and consumer.
+
+=cut
+
 # TODO retest legacy parameters against someone who has them
 sub hats {
     (
         qw/auth parameters/,
         fb11authdb => {
             class => 'auth'
-        }
+        },
+        'dbicdh::consumer' => {
+            class => '+OpusVL::FB11::Hat::dbicdh::consumer::is_brain',
+            constructor => {
+                priority => 0,
+            }
+        },
+        'dbicdh::manager',
     )
 }
 
 sub provided_services {
-    qw/auth fb11authdb/
+    qw/auth fb11authdb dbicdh::manager/
+}
+
+# TODO: I don't like this being done automatically but we have to make sure DBs
+# are deployed before other inits run
+sub hive_init {
+    my $self = shift;
+    my $hive = shift;
+
+    $hive->service('dbicdh::manager')->deploy_and_upgrade($hive);
+
+    # XXX TEMPORARY
+    # This is not where this should live but this is sort of the right place.
+    # The correct behaviour would be a UI that does this on request, which we
+    # can direct to if we have no auth data yet.
+    $self->_setup_fb11admin($hive);
+}
+
+sub _setup_fb11admin {
+    my $self = shift;
+    my $hive = shift;
+
+    # If we don't have an app, just don't do it. This only matters if there's an
+    # app to provide auth to (although why are you here if there's no app?)
```

Comment:
Why not?  The hive system is pretty generic and its use isn't predicated on there being a web interface, and I'm pretty sure the auth system could be used without an app too even if only in tests.

Also what about public websites that don't need an auth?

Should also consider (even if we don't handle it immediately) the possibility that there could be two different sets of users authed in different ways - e.g. customers for the e-commerce website, and LDAP users for staff for the backend.

Reply:
Ah of course I see the app needs to be there for auth_feature_list to exist.  Will do for now but will need to consider another way when we go generic.

Reply:
Agreed; this hat currently is designed to stitch together the auth-heavy FB11 in webapp mode with the feature-agnostic FB11 in hive mode.

Actually implementing alternative (or no) auth is a discussion for the future, and I'm not sure if it's one we want to have before FB11 v1.0 or not.

I also think authentication and authorisation (the ACL) should be separate services, since there's absolutely no reason why we should expect, or force, our stash of users to know what those users can access.

Reply:
Several issues are contingent on this so we'll get back to this later.